dlt/pipeline/state_sync.py

```diff
@@ -52,6 +53,27 @@
 }
 
 
+def json_encode_state(state: TPipelineState) -> str:
+    return json.typed_dumps(state)
+
+
+def json_decode_state(state_str: str) -> DictStrAny:
+    return json.typed_loads(state_str)  # type: ignore[no-any-return]
+
+
+def compress_state(state: TPipelineState) -> str:
+    return compressed_b64encode(json.typed_dumpb(state))
+
+
+def decompress_state(state_str: str) -> DictStrAny:
+    try:
+        state_bytes = compressed_b64decode(state_str)
+    except binascii.Error:
+        return json.typed_loads(state_str)  # type: ignore[no-any-return]
+    else:
+        return json.typed_loadb(state_bytes)  # type: ignore[no-any-return]
+
+
 def merge_state_if_changed(old_state: TPipelineState, new_state: TPipelineState, increase_version: bool = True) -> Optional[TPipelineState]:
     # we may want to compare hashes like we do with schemas
     if json.dumps(old_state, sort_keys=True) == json.dumps(new_state, sort_keys=True):
```

Comment:
here state is dumped but just to compare the content. we are good with a regular encoder